<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MeshEditing</class>
 <widget class="QWidget" name="MeshEditing">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>384</width>
    <height>501</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MeshEditing</string>
  </property>
  <widget class="QToolBox" name="toolBox">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>10</y>
     <width>381</width>
     <height>481</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="page">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>381</width>
      <height>427</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Page 1</string>
    </attribute>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>150</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>angle:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>110</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>axis:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="rotateZ_2">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>110</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="rotateAngle_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>150</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="rotateY_2">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>110</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="scaleX_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="translateX_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>230</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="translateZ_2">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>230</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Translation:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="scaleY_2">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="translateY_2">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>230</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="rotateX_2">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>110</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-10.000000000000000</double>
     </property>
     <property name="maximum">
      <double>10.990000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="scaleZ_2">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotation:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Scale:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="update">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>270</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>update</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>381</width>
      <height>427</height>
     </rect>
    </property>
    <attribute name="label">
     <string>Page 2</string>
    </attribute>
    <widget class="QComboBox" name="submesh">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>151</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="renderunit">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>201</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>MeshRenderUnit:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>46</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>SubMesh:</string>
     </property>
    </widget>
    <widget class="QTabWidget" name="submeshTab">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>331</width>
       <height>321</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="material">
      <attribute name="title">
       <string>Tab 1</string>
      </attribute>
      <widget class="QLabel" name="spc">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>50</y>
         <width>81</width>
         <height>20</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QPushButton" name="ambientC">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>20</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Ambient</string>
       </property>
      </widget>
      <widget class="QPushButton" name="diffuseC">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Diffuse</string>
       </property>
      </widget>
      <widget class="QLabel" name="amb">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>50</y>
         <width>81</width>
         <height>20</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="dif">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>81</width>
         <height>20</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QPushButton" name="specularC">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>20</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Specular</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="transform">
      <attribute name="title">
       <string>Tab 2</string>
      </attribute>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Scale:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scaleX">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>40</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scaleY">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scaleZ">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>40</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Rotation:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotateX">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>110</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotateY">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>110</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotateZ">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>110</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>110</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>axis:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>150</y>
         <width>46</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>angle:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="rotateAngle">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>150</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>61</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Translation:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="translateX">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>230</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="translateZ">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>230</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="translateY">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>230</y>
         <width>62</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-10.000000000000000</double>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
